server/src/main/java/io/druid/segment/indexing/TuningConfig.java

```diff
@@ -33,4 +33,9 @@
   boolean DEFAULT_LOG_PARSE_EXCEPTIONS = false;
   int DEFAULT_MAX_PARSE_EXCEPTIONS = Integer.MAX_VALUE;
   int DEFAULT_MAX_SAVED_PARSE_EXCEPTIONS = 0;
+  int DEFAULT_MAX_ROWS_IN_MEMORY = 1_000_000;
+  // We initially estimated this to be 1/3(max jvm memory), but bytesCurrentlyInMemory only
+  // tracks active index and not the index being flushed to disk, to account for that
+  // we halved default to 1/6(max jvm memory)
+  long DEFAULT_MAX_BYTES_IN_MEMORY = Runtime.getRuntime().maxMemory() / 6;
```

Comment:
I think it's worthwhile to roughly comment about this lazy initialization in each tuningConfig implementation including a link to here.

Reply:
Not clear where should this comment be present. But i have added it where it's initially initialized to 0.